libs/core/tests/unit/string_tests.cpp

```diff
@@ -17,12 +17,23 @@
 //------------------------------------------------------------------------------
 
 #include "core/string/ends_with.hpp"
+#include "core/string/replace.hpp"
 #include "core/string/starts_with.hpp"
+#include "core/string/to_lower.hpp"
+#include "core/string/trim.hpp"
 
 #include "gtest/gtest.h"
 
 using namespace fetch::core;
 
+template <class T, class U>
+inline bool Equal(T &&t, U &&u)
+{
+  return t == u;
+}
+
+#define EXPECT_EQUAL(...) EXPECT_TRUE(Equal(__VA_ARGS__))
```

Comment:
Let's not reinvent the wheel. This code is longer than what the check_replace test could be if you simply used gtest macros. Gtest not handling spaces in assertion macros is limitation that's more easily circumvented by extracting macro arguments to a suitably named local constant. In general I'd be careful with trying to shoehorn as much code as possible into every line (I'm referring to `check_Replace` here).

Reply:
> Gtest not handling spaces
It's commas you mean.